### PR TITLE
Fix/cfds 59 url input validation criteria

### PR DESCRIPTION
**Pull request by SamerIssam**
## Description

Operator Details input now accept https:// or http://  as url 

## Testing instructions

Submit a Operator Details form

### CODE REVIEW

File: repos/fdbt-site/src/pages/api/manageOperatorDetails.ts [hunk 1]
```diff
@@ -28,9 +28,10 @@ export const collectErrors = (operatorDetails: OperatorDetails): ErrorInfo[] =>
     Object.entries(operatorDetails)
         .filter((entry) => {
             const inputValue = entry[1];
+            const inputKey = entry[0];
 
-            const entryHasInvalidCharacters = invalidCharactersArePresent(inputValue);
-
+            const entryHasInvalidCharacters =
+                inputKey === 'url' ? invalidUrlInput(inputValue) : invalidCharactersArePresent(inputValue);
```
Comment (RobCattonKPMG) [inline repos/fdbt-site/src/pages/api/manageOperatorDetails.ts:34]: I think we should be leaving this chunk of code as it was, and using 'invalidUrlInput' function on line 56 instead.

File: repos/fdbt-site/src/pages/api/manageOperatorDetails.ts [hunk 2]
```diff
@@ -52,7 +53,6 @@ export const collectErrors = (operatorDetails: OperatorDetails): ErrorInfo[] =>
     if (operatorDetails.email && !checkEmailValid(operatorDetails.email)) {
         errors.push({ id: 'email', errorMessage: 'Provide a valid email' });
     }
-
     if (operatorDetails.url && !/^[^ ]+$/.exec(operatorDetails.url)) {
```
Comment (RobCattonKPMG) [inline repos/fdbt-site/src/pages/api/manageOperatorDetails.ts:56]: This is where we should remove `!/^[^ ]+$/.exec(operatorDetails.url)` and replace with your function.

File: repos/fdbt-site/src/utils/apiUtils/validator.ts [hunk 3]
```diff
@@ -42,6 +42,12 @@ export const invalidCharactersArePresent = (value: string): boolean => {
 
     return regularExpression.test(value);
 };
+export const invalidUrlInput = (value: string): boolean => {
```
Comment (RobCattonKPMG) [inline repos/fdbt-site/src/utils/apiUtils/validator.ts:45]: Could we add some unit tests for this function with some different inputs, to show that it works for what we expect (www., https:// etc) and rejects nonUrls.

File: repos/fdbt-site/src/pages/api/manageOperatorDetails.ts (hunk 1 above)
Comment (SamerIssam) [reply repos/fdbt-site/src/pages/api/manageOperatorDetails.ts:34]: we have to have something to we stop  passing the url key to the `nvalidCharactersArePresent` function.

**Review by RobCattonKPMG — APPROVED** [review]

**Pull request merged**